engine/classes/Elgg/FileService/File.php

```diff
@@ -106,6 +108,12 @@ public function getURL() {
 			return false;
 		}
 
+		if (preg_match('~[^a-zA-Z0-9_\./ ]~', $relative_path)) {
```

Comment:
I am sure we will dealing with some hickups in the future, so as long as we get most of the craziness covered, I can live with it.